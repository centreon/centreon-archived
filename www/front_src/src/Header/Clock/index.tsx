import * as React from 'react';

import { Typography, makeStyles } from '@material-ui/core';

import { useLocaleDateTimeFormat } from '@centreon/ui';

const useStyles = makeStyles((theme) => ({
  dateTime: {
    color: theme.palette.common.white,
  },
}));

const Clock = (): JSX.Element => {
  const classes = useStyles();

  const refreshIntervalRef = React.useRef<number>();
  const [dateTime, setDateTime] = React.useState({
    date: '',
    time: '',
  });

  const { format, toTime } = useLocaleDateTimeFormat();

  const updateDateTime = (): void => {
    const now = new Date();

    const date = format({ date: now, formatString: 'LL' });
    const time = toTime(now);

    setDateTime({ date, time });
  };

  React.useEffect(() => {
    updateDateTime();

    const thirtySeconds = 30 * 1000;
    refreshIntervalRef.current = window.setInterval(
      updateDateTime,
      thirtySeconds,
    );

    return (): void => {
      clearInterval(refreshIntervalRef.current);
    };
  }, []);

  const { date, time } = dateTime;

  return (
    <div className={classes.dateTime}>
      <Typography variant="body2">{date}</Typography>
      <Typography variant="body1">{time}</Typography>
    </div>
  );
};

export default Clock;
