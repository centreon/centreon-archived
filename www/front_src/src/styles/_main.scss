@import "partials/typography";
@import "partials/fonts";
@import "partials/variables";
@import "partials/messages";
@import "partials/icons";
@import "partials/form/form";
@import "partials/helpers";
@import "partials/mixins";
