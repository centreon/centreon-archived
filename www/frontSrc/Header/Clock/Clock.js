import React from 'react'
import { withStyles } from 'material-ui/styles'
import Avatar from 'material-ui/Avatar'
import { MenuItem } from 'material-ui/Menu'
import IconButton from 'material-ui/IconButton'
import Grid from 'material-ui/Grid'
import Popover from 'material-ui/Popover'
import Button from 'material-ui/Button'
import Bookmark from 'material-ui-icons/Bookmark'
import VolumeUp from 'material-ui-icons/VolumeUp'
import Typography from 'material-ui/Typography'

const styles = theme => ({
  profileRoot: {
    display: 'flex',
    flexDirection: 'row-reverse',
    fontFamily: theme.font.openSans,
  },
  moment: {
    alignSelf: 'flex-start',
    margin: '10px 0px',
  },
  headerDate: {
    fontSize: 14,
  },
  headerTime: {
    fontSize: 35,
    lineHeight: '28px',
  },
})

const Clock = ({
  classes,
    currentDate,
  }) => (
      <div className={classes.moment}>
        <div className={classes.headerDate}> {currentDate.date} </div>
        <div className={classes.headerTime}> {currentDate.time} </div>
      </div>
    )

export default withStyles(styles)(Clock)
