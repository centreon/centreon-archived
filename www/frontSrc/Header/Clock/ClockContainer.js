import React, { Component } from 'react'
import { connect } from 'react-redux'
import Clock from './Clock'
import { getClock } from "../../webservices/clockApi"
import 'moment-timezone'
import Moment from 'moment'

class ClockContainer extends Component {

  constructor(props) {
    super(props)
    this.state = {
      currentDate: {},
    }
  }

  componentWillReceiveProps(nextProps) {
    if (this.props !== nextProps) {
      this.setState({
        currentDate: this.setDate(nextProps.clock.timezone, nextProps.clock.locale, nextProps.clock.time)
      })
    }
  }

  componentDidMount = () =>  {
    this.props.getClock()

  }

  setDate = (tz, locale, timestamp) => {

    const currentTime = tz !== '' ? Moment.unix(timestamp).tz(tz) : Moment.unix(timestamp)
    locale !== null ? currentTime.locale(locale) : currentTime

    return {
      date: currentTime.format('LL'),
      time: currentTime.format('LT')
    }
  }

  render () {
    const { currentDate } = this.state

    return (
      <Clock
        currentDate={currentDate}
      />
    )
  }
}

const mapStateToProps = (store) => {
  return {
    clock: store.clock.data,
  }
}

const mapDispatchToProps = (dispatch) => {
  return {
    getClock: () => {
      return dispatch(getClock())
    },
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(ClockContainer)