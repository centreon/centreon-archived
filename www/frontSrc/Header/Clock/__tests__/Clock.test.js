/* eslint-env jest */
import React from 'react'
import Enzyme from 'enzyme'
import { shallow, mount } from 'enzyme'
import toJson from 'enzyme-to-json'
import Adapter from 'enzyme-adapter-react-16'
import Clock from '../ClockContainer'
import configureStore from 'redux-mock-store'
import thunk from 'redux-thunk'
import { MuiThemeProvider, createMuiTheme } from 'material-ui/styles'

Enzyme.configure({ adapter: new Adapter() })

const theme = createMuiTheme({
  font: {
    openSans: "'Open Sans', Arial, Tahoma, Helvetica, Sans-Serif"
  },
})

describe('ClockObject', () => {
  const middlewares = [ thunk ]
  const mockStore = configureStore(middlewares)

  const store = mockStore({
    clock: {
      dataFetched: true,
      date: 1525425186,
      isFetching: false,
      locale: "en_US",
      refreshTime: 120000,
      time: 1525425186,
      timezone: "America/Argentina/Salta",
    }
  })

  let getClock = jest.fn()

  it('should display time and date', () => {
    const component = mount(
      <MuiThemeProvider theme={theme}>
        <Clock store={store}
               getClock={getClock} />
      </MuiThemeProvider>
    )

    const actions = store.getActions()
    expect(actions[0].type).toBe('REQUEST_CLOCK')
  })

  it('Clock renders correctly', () => {
    const component = shallow(<Clock store={store} />)
    expect(toJson(component)).toMatchSnapshot()
  })

  it('clock function should be called', () => {
    const component = shallow(<Clock store={store} />).dive()
    const instance = component.instance()

    spyOn(instance, 'setDate').and.callThrough()
    expect(instance.setDate('America/Argentina/Salta','en_US', 1525425186))
    expect(instance.setDate).toBeTruthy()
  })
})