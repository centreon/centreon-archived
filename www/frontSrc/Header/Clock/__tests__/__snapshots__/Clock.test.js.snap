// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClockObject Clock renders correctly 1`] = `
<ClockContainer
  clock={
    Object {
      "dataFetched": true,
      "date": 1525425186,
      "isFetching": false,
      "locale": "en_US",
      "refreshTime": 120000,
      "time": 1525425186,
      "timezone": "America/Argentina/Salta",
    }
  }
  getClock={[Function]}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
  storeSubscription={
    Subscription {
      "listeners": Object {
        "clear": [Function],
        "get": [Function],
        "notify": [Function],
        "subscribe": [Function],
      },
      "onStateChange": [Function],
      "parentSub": undefined,
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
      "unsubscribe": [Function],
    }
  }
  timeDispatcher={[Function]}
/>
`;
